deepmedic/frontEnd/trainSession.py

```diff
@@ -19,148 +19,186 @@
                     self._log.print3("Loading trainer parameters...")
                     saver_trainer.restore(sessionTf, chkpt_fname)
                     self._log.print3("Trainer parameters were loaded.")
                 else:
                     self._log.print3("Reset of trainer parameters was requested. Re-initializing them...")
-                    tf.variables_initializer(var_list = collection_vars_trainer).run()
+                    tf.variables_initializer(var_list=collection_vars_trainer).run()
                     self._log.print3("Trainer parameters re-initialized.")
-            else :
+            else:
                 self._log.print3("=========== Initializing network and trainer variables  ===============")
-                # tf.variables_initializer(var_list = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES) ).run() # Initializes all.
+                # Initializes all.
+                # tf.variables_initializer(var_list = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES) ).run()
                 # Initialize separate as below, so that in case I miss a variable, I will get an error and I will know.
-                tf.variables_initializer(var_list = collection_vars_net).run()
-                tf.variables_initializer(var_list = collection_vars_trainer).run()
+                tf.variables_initializer(var_list=collection_vars_net).run()
+                tf.variables_initializer(var_list=collection_vars_trainer).run()
                 self._log.print3("All variables were initialized.")
-                
+
                 filename_to_save_with = self._params.filepath_to_save_models + ".initial." + datetimeNowAsStr()
                 self._log.print3("Saving the initial model at:" + str(filename_to_save_with))
-                saver_all.save( sessionTf, filename_to_save_with+".model.ckpt", write_meta_graph=False )
-                # tf.train.write_graph( graph_or_graph_def=sessionTf.graph.as_graph_def(), logdir="", name=filename_to_save_with+".graph.pb", as_text=False)
-             
+                saver_all.save(sessionTf, filename_to_save_with+".model.ckpt", write_meta_graph=False)
+                # tf.train.write_graph( graph_or_graph_def=sessionTf.graph.as_graph_def(),
+                # logdir="", name=filename_to_save_with+".graph.pb", as_text=False)
+
             self._log.print3("")
             self._log.print3("=======================================================")
             self._log.print3("============== Training the CNN model =================")
             self._log.print3("=======================================================\n")
-            
-            res_code = do_training( *( [sessionTf, saver_all, cnn3d, trainer] + self._params.get_args_for_train_routine() ) )
-            
-            
+
+            do_training(*([sessionTf, saver_all, cnn3d, trainer] + self._params.get_args_for_train_routine()),
+                        tensorboard_loggers=tensorboard_loggers)
```

Comment:
I would move it within the [...], which holds all the tensorflow parameters:

[sessionTf, saver_all, cnn3d, trainer, tensorboard_loggers]

Reply:
I think we should make a class that holds all the parameters and parse that, but that's a bigger structural task. I'd leave it for the

Reply:
you mean because it has an optional arg